### ISSUE TITLE
[bug] `make.bat epub` | WARNING: cannot read image file ... copying it instead; unkown mime type for ...

### ISSUE THREAD

**Issue by nandgator** [sphinx-doc/sphinx#10387]
### Describe the bug

I'm including an image directive:

```rst
|python|

..
    ...

.. |python| image:: https://img.shields.io/badge/python-3.10%20%7C%203.11-red
    :alt: Python Versions
```

It should render an badge ![](https://img.shields.io/badge/python-3.10%20%7C%203.11-red), which it does well, in HTML builds, but fails, while generating ePUB.

### How to Reproduce

1. Add a simple badge anywhere in the project
2. Build ePUB file

It retorts to warning..

```py
...

WARNING: cannot read image file 'path\\to\\file': copying it instead

...

WARNING: unknown mimetype for _images\file, ignoring

...

```

### Expected behavior

The badge should render in ePUB file w/o any issue.

### Your project

> Sorry, can't share

### Screenshots

![image](https://user-images.githubusercontent.com/38415384/165062559-8fc55244-6185-421f-a855-91fe4f5e9066.png)

### OS

Win10

### Python version

3.10

### Sphinx version

4.5

### Sphinx extensions

`sphinx.ext.duration`

### Additional context

I've noticed the file extension `.svg` is not in the path. I suspect the behavior has to do with the `.` character being present in the URL.

**Comment by AA-Turner** [comment]
Adding `.svg` to the URL seems to work as a workaround.

A

**Comment by joe733** [comment]
No, it does not. Appending `.svg` does not help. Instead removing:
-  `.` from version numbers `3.10`, `3.11` and
-  the pipe character `|` (`%7C`) and
- the spaces  (`%20`)

works.. but that's a bad workaround.

**Comment by AA-Turner** [comment]
This worked for me locally, but I'll re-open & investigate further.

A
